decode.go

```diff
@@ -46,16 +46,16 @@ func (l *Lamenv) decode(conf reflect.Value, parts []string) error {
 
 	if p, ok := ptr.Interface().(encoding.TextUnmarshaler); ok {
 		if variable, input, exist := lookupEnv(parts); exist {
-			// remove the variable to avoid to reuse it later
+			// remove the variable to avoid reusing it later
 			delete(l.env, variable)
 			if err := p.UnmarshalText([]byte(input)); err != nil {
 				return err
 			}
 
 			// in case the method UnmarshalEnv() is setting some parameter in the struct, we have to save these changes
 			v.Set(ptr.Elem())
-			return nil
 		}
+		return nil
```

Comment:
Isn't there the same problem in UnmarshalEnv section?

Reply:
actually no, if you look closer, the `return nil` is at the correct place. So should be good

Reply:
nevermind

Reply:
Yeah I checked after, I thought it was a copy paste :P